internal/controller/install/executor_controller.go

```diff
@@ -76,17 +76,14 @@ type ExecutorReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *ExecutorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	logger := log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("kind", "Executor", "namespace", req.Namespace, "name", req.Name)
+
 	started := time.Now()
-	logger.Info("Reconciling Executor object")
 
-	logger.Info("Fetching Executor object from cache")
+	logger.Info("Reconciling object")
+
 	var executor installv1alpha1.Executor
-	if err := r.Client.Get(ctx, req.NamespacedName, &executor); err != nil {
-		if k8serrors.IsNotFound(err) {
-			logger.Info("Executor not found in cache, ending reconcile")
-			return ctrl.Result{}, nil
-		}
+	if miss, err := getObjectFromCache(ctx, r.Client, &executor, req.NamespacedName, logger); err != nil || miss {
```

Comment:
How about this:
```go
if err := r.Get(ctx, req.NamespacedName, &executor); err != nil {
	return ctrl.Result{}, client.IgnoreNotFound(err)
}
```

Reply:
Reason why I extracted it is to log behaviour in a uniform way and to be reused in all controllers as I expect the same thing to happen in each controller.

Reason for this cleanups is that similar logic was implemented multiple times in each controller and that led to many code duplication.